sound/soc/intel/boards/sof_sdw.c

```diff
@@ -682,6 +683,7 @@ static struct sof_sdw_codec_info codec_info_list[] = {
 				.dailink = {SDW_JACK_OUT_DAI_ID, SDW_JACK_IN_DAI_ID},
 				.init = sof_sdw_rt_sdca_jack_init,
 				.exit = sof_sdw_rt_sdca_jack_exit,
+				.rtd_init = rt_sdca_jack_rtd_init,
```

Comment:
sorry, I am not following what the change is.

we had a perfectly symmetric solution with init/exit, but now init is replaced by rtd_init, should we have an rtd_exit?

This looks rather odd for now, I don't know what to make of this.

Reply:
> sorry, I am not following what the change is.
> 
> we had a perfectly symmetric solution with init/exit, but now init is replaced by rtd_init, should we have an rtd_exit?


The init/exit was not perfectly symmetric for example, there is a [.init = sof_sdw_rt700_init,](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/intel/boards/sof_sdw.c#L669), and no sof_sdw_rt700_exit. And the only thing [sof_sdw_rt700_init](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/intel/boards/sof_sdw_rt700.c#L113) does is to assign `dai_links->init = rt700_rtd_init;`.
The PR removes `dai_links->init = xxx_rtd_init;` from the .init function, and run `xxx_rtd_init` in `sof_sdw_rtd_init`. 
We don't have dai_link.exit callback for now. But, yes, we probably, need a `sof_sdw_rtd_exit` to call rtd_exit in case some codecs need it someday.

> 
> This looks rather odd for now, I don't know what to make of this.